--- ./MoinMoin/auth/__init__.py.orig	2016-10-31 22:44:02.000000000 +0200
+++ ./MoinMoin/auth/__init__.py	2016-11-28 15:26:43.662260442 +0200
@@ -371,7 +371,7 @@ class GivenAuth(BaseAuth):
             auth_username = self.transform_username(auth_username)
             logging.debug("auth_username (after decode/transform) = %r" % auth_username)
             u = user.User(request, auth_username=auth_username,
-                          auth_method=self.name, auth_attribs=('name', 'password'))
+                          auth_method=self.name, auth_attribs=('name'))
 
         logging.debug("u: %r" % u)
         if u and self.autocreate:
