--- ./MoinMoin/config/__init__.py.orig	2016-10-31 22:44:02.000000000 +0200
+++ ./MoinMoin/config/__init__.py	2016-11-28 15:26:43.663260442 +0200
@@ -25,7 +25,7 @@ umask = 0770
 # list of acceptable password hashing schemes for cfg.password_scheme,
 # here we only give reasonably good schemes, which is passlib (if we
 # have passlib) and ssha (if we only have builtin stuff):
-password_schemes_configurable = ['{PASSLIB}', '{SSHA}', ]
+password_schemes_configurable = ['{PASSLIB}', '{SSHA}', '{SHA}' ]
 
 # ordered list of supported password hashing schemes, best (passlib) should be
 # first, best builtin one should be second. this is what we support if we
@@ -58,6 +58,9 @@ page_invalid_chars_regex = re.compile(
     ur"""
     \u0000 | # NULL
 
+    \# | # http://tools.ietf.org/html/rfc3986#section-3.3
+    \? |
+
     # Bidi control characters
     \u202A | # LRE
     \u202B | # RLE
