--- ./MoinMoin/theme/__init__.py.orig	2014-10-17 22:45:32.000000000 +0300
+++ ./MoinMoin/theme/__init__.py	2014-10-20 11:53:32.884284973 +0300
@@ -48,6 +48,7 @@
         'diff':        (_("Diffs"),               "moin-diff.png",   15, 11),
         'info':        (_("Info"),                "moin-info.png",   12, 11),
         'edit':        (_("Edit"),                "moin-edit.png",   12, 12),
+        'formedit':    (_("FormEdit"),            "moin-news.png",     12, 12),
         'unsubscribe': (_("Unsubscribe"),         "moin-unsubscribe.png", 14, 10),
         'subscribe':   (_("Subscribe"),           "moin-subscribe.png", 14, 10),
         'raw':         (_("Raw"),                 "moin-raw.png",    12, 13),
