--- ./MoinMoin/theme/__init__.py.orig	2016-10-31 22:44:02.000000000 +0200
+++ ./MoinMoin/theme/__init__.py	2016-11-28 15:26:43.676260441 +0200
@@ -46,6 +46,7 @@ class ThemeBase:
         'diff':        (_("Diffs"),               "moin-diff.png",   15, 11),
         'info':        (_("Info"),                "moin-info.png",   12, 11),
         'edit':        (_("Edit"),                "moin-edit.png",   12, 12),
+        'formedit':    (_("FormEdit"),            "moin-news.png",     12, 12),
         'unsubscribe': (_("Unsubscribe"),         "moin-unsubscribe.png", 14, 10),
         'subscribe':   (_("Subscribe"),           "moin-subscribe.png", 14, 10),
         'raw':         (_("Raw"),                 "moin-raw.png",    12, 13),
