--- ./MoinMoin/web/contexts.py.orig	2016-10-31 22:44:02.000000000 +0200
+++ ./MoinMoin/web/contexts.py	2016-11-28 15:26:43.681260441 +0200
@@ -218,6 +218,12 @@ class HTTPContext(BaseContext):
 
     # proxy further attribute lookups to the underlying request first
     def __getattr__(self, name):
+        if name == 'editlog':
+            if "editlog" not in self.__dict__:
+                from MoinMoin.logfile import editlog
+                self.request.rootpage = self.rootpage
+                self.editlog = editlog.EditLog(self.request)
+            return self.editlog
         try:
             return getattr(self.request, name)
         except AttributeError, e:
