--- ./MoinMoin/auth/__init__.py.orig	2014-10-17 22:45:32.000000000 +0300
+++ ./MoinMoin/auth/__init__.py	2014-10-20 11:53:32.869284981 +0300
@@ -371,7 +371,7 @@
             auth_username = self.transform_username(auth_username)
             logging.debug("auth_username (after decode/transform) = %r" % auth_username)
             u = user.User(request, auth_username=auth_username,
-                          auth_method=self.name, auth_attribs=('name', 'password'))
+                          auth_method=self.name, auth_attribs=('name'))
 
         logging.debug("u: %r" % u)
         if u and self.autocreate:
