--- ./MoinMoin/auth/http.py.orig   2010-12-15 12:01:30.719630611 +0200
+++ ./MoinMoin/auth/http.py        2010-12-15 12:03:21.269298026 +0200
@@ -84,7 +84,7 @@ class HTTPAuth(BaseAuth):
                 # when using http auth, we have external user name and password,
                 # we don't use the moin user profile for those attributes.
                 u = user.User(request, auth_username=username,
-                              auth_method=self.name, auth_attribs=('name', 'password'))
+                              auth_method=self.name, auth_attribs=('name'))
 
         logging.debug("u: %r" % u)
         if u and self.autocreate:

