--- ./wiki/htdocs/common/js/common.js.orig	2010-06-05 19:14:35.000000000 +0300
+++ ./wiki/htdocs/common/js/common.js	2012-01-23 13:48:50.727695002 +0200
@@ -98,9 +98,6 @@ function can_use_gui_editor() {
     if ( window.opera && window.opera.version && parseFloat( window.opera.version() ) >= 9.5 )
         return true ;
 
-/*
-  // disable safari : until fck devteam fix http://dev.fckeditor.net/ticket/2333
-  
     // Adobe AIR
     // Checked before Safari because AIR have the WebKit rich text editor
     // features from Safari 3.0.4, but the version reported is 420.
@@ -109,8 +106,11 @@ function can_use_gui_editor() {
 
     // Safari 3+
     if ( sAgent.indexOf( ' applewebkit/' ) != -1 )
-        return ( sAgent.match( / applewebkit\/(\d+)/ )[1] >= 522 ) ;    // Build must be at least 522 (v3)
-*/
+	if ( sAgent.indexOf( ' chrome/' ) != -1 )
+	    return ( sAgent.match( / chrome\/(\d+)/ )[1] >= 15 ) ;    // Chrome/Chromium version must be at least 15.x
+	else
+	    return ( sAgent.match( / applewebkit\/(\d+)/ )[1] >= 522 ) ;    // Build must be at least 522 (v3)
+
     return false ;
 
 }
