--- ./MoinMoin/config/__init__.py.orig	2014-10-17 22:45:32.000000000 +0300
+++ ./MoinMoin/config/__init__.py	2014-10-20 11:53:32.870284976 +0300
@@ -25,7 +25,7 @@
 # list of acceptable password hashing schemes for cfg.password_scheme,
 # here we only give reasonably good schemes, which is passlib (if we
 # have passlib) and ssha (if we only have builtin stuff):
-password_schemes_configurable = ['{PASSLIB}', '{SSHA}', ]
+password_schemes_configurable = ['{PASSLIB}', '{SSHA}', '{SHA}' ]
 
 # ordered list of supported password hashing schemes, best (passlib) should be
 # first, best builtin one should be second. this is what we support if we
