--- ./wiki/htdocs/applets/moinFCKplugins/selection/fckplugin.js.orig	2011-11-28 22:40:58.000000000 +0200
+++ ./wiki/htdocs/applets/moinFCKplugins/selection/fckplugin.js	2012-07-29 22:11:17.602690454 +0300
@@ -321,6 +321,8 @@ else {
   FCKSelection.GetStartNode = function()
     {
       var oSelection = FCKSelection.GetSelection();
+      if (oSelection.rangeCount == 0) // this fixes Chrome Browser exception on FCKeditor init
+        return null;
       // startContainer returns the parent of the first node in the selection
       var oContainer = oSelection.getRangeAt(0).startContainer;
       // The offset within the startContainer where the range starts
@@ -343,6 +345,8 @@ else {
   FCKSelection.GetEndNode = function()
   {
     var oSelection = FCKSelection.GetSelection();
+    if (oSelection.rangeCount == 0) // this fixes Chrome Browser exception on FCKeditor init
+      return null;
     var oContainer = oSelection.getRangeAt(0).endContainer;
     var iOffset = oSelection.getRangeAt(0).startEndset;
 
