--- ./MoinMoin/Page.py.orig	2012-03-13 01:12:35.665826503 +0200
+++ ./MoinMoin/Page.py	2012-03-13 01:13:35.365909957 +0200
@@ -108,8 +108,7 @@ class ItemCache:
             (for 'meta') or the complete cache ('pagelists').
             @param request: the request object
         """
-        from MoinMoin.logfile import editlog
-        elog = editlog.EditLog(request)
+        elog = request.editlog
         old_pos = self.log_pos
         new_pos, items = elog.news(old_pos)
         if items:
