--- ./MoinMoin/theme/modernized.py-orig	2010-06-05 19:14:34.000000000 +0300
+++ ./MoinMoin/theme/modernized.py	2011-11-21 17:56:05.435819000 +0200
@@ -20,6 +20,8 @@ class Theme(ThemeBase):
         # FileAttach
         'attach':     ("%(attach_count)s",       "moin-attach.png",   16, 16),
         'info':       ("[INFO]",                 "moin-info.png",     16, 16),
+        'edit':       (_("Edit"),                "moin-edit.png",     12, 12),
+        'formedit':   (_("FormEdit"),            "moin-news.png",     12, 12),
         'attachimg':  (_("[ATTACH]"),            "attach.png",        32, 32),
         # RecentChanges
         'rss':        (_("[RSS]"),               "moin-rss.png",      16, 16),
