diff -up ejabberd-2.1.11/src/extauth.erl.orig ejabberd-2.1.11/src/extauth.erl
--- ejabberd-2.1.11/src/extauth.erl.orig	2012-04-26 17:04:16.000000000 +0300
+++ ejabberd-2.1.11/src/extauth.erl	2012-07-17 12:55:07.000000000 +0300
@@ -35,7 +35,8 @@
 	 try_register/3,
 	 remove_user/2,
 	 remove_user/3,
-	 is_user_exists/2]).
+	 is_user_exists/2,
+     get_affiliation/4]).
 
 -include("ejabberd.hrl").
 
@@ -95,6 +96,9 @@ remove_user(User, Server) ->
 remove_user(User, Server, Password) ->
     call_port(Server, ["removeuser3", User, Server, Password]).
 
+get_affiliation(User, Server, Room, RoomServer) ->
+    call_port(?MYNAME, ["getaff", User, Server, Room, RoomServer]).
+
 call_port(Server, Msg) ->
     LServer = jlib:nameprep(Server),
     ProcessName = get_process_name(LServer, random_instance(get_instances(LServer))),
@@ -169,5 +173,7 @@ encode(L) ->
 decode([0,0]) ->
     false;
 decode([0,1]) ->
-    true.
+    true;
+decode(String) ->
+    list_to_atom(String).
 
diff -up ejabberd-2.1.11/src/mod_muc/mod_muc_room.erl.orig ejabberd-2.1.11/src/mod_muc/mod_muc_room.erl
--- ejabberd-2.1.11/src/mod_muc/mod_muc_room.erl.orig	2012-04-26 17:04:16.000000000 +0300
+++ ejabberd-2.1.11/src/mod_muc/mod_muc_room.erl	2012-07-17 12:55:07.000000000 +0300
@@ -1287,43 +1287,14 @@ set_affiliation(JID, Affiliation, StateD
     StateData#state{affiliations = Affiliations}.
 
 get_affiliation(JID, StateData) ->
-    {_AccessRoute, _AccessCreate, AccessAdmin, _AccessPersistent} = StateData#state.access,
-    Res =
-	case acl:match_rule(StateData#state.server_host, AccessAdmin, JID) of
-	    allow ->
-		owner;
-	    _ ->
-		LJID = jlib:jid_tolower(JID),
-		case ?DICT:find(LJID, StateData#state.affiliations) of
-		    {ok, Affiliation} ->
-			Affiliation;
-		    _ ->
-			LJID1 = jlib:jid_remove_resource(LJID),
-			case ?DICT:find(LJID1, StateData#state.affiliations) of
-			    {ok, Affiliation} ->
-				Affiliation;
-			    _ ->
-				LJID2 = setelement(1, LJID, ""),
-				case ?DICT:find(LJID2, StateData#state.affiliations) of
-				    {ok, Affiliation} ->
-					Affiliation;
-				    _ ->
-					LJID3 = jlib:jid_remove_resource(LJID2),
-					case ?DICT:find(LJID3, StateData#state.affiliations) of
-					    {ok, Affiliation} ->
-						Affiliation;
-					    _ ->
-						none
-					end
-				end
-			end
-		end
-	end,
-    case Res of
-	{A, _Reason} ->
-	    A;
-	_ ->
-	    Res
+    {User, Server, _Resource} = jlib:jid_tolower(JID),
+    Room = StateData#state.room,
+    RoomServer = StateData#state.host,
+    case extauth:get_affiliation(User, Server, Room, RoomServer) of
+        false ->
+            none;
+        Affiliation -> 
+            Affiliation
     end.
 
 get_service_affiliation(JID, StateData) ->
diff -up ejabberd-2.1.11/src/mod_muc/mod_muc_room.hrl.orig ejabberd-2.1.11/src/mod_muc/mod_muc_room.hrl
--- ejabberd-2.1.11/src/mod_muc/mod_muc_room.hrl.orig	2012-04-26 17:04:16.000000000 +0300
+++ ejabberd-2.1.11/src/mod_muc/mod_muc_room.hrl	2012-07-17 12:55:07.000000000 +0300
@@ -39,8 +39,8 @@
 		 persistent = false,
 		 moderated = true,
 		 captcha_protected = false,
-		 members_by_default = true,
-		 members_only = false,
+		 members_by_default = false,
+		 members_only = true,
 		 allow_user_invites = false,
 		 password_protected = false,
 		 password = "",
diff -up ejabberd-2.1.11/src/web/ejabberd_http.erl.orig ejabberd-2.1.11/src/web/ejabberd_http.erl
--- ejabberd-2.1.11/src/web/ejabberd_http.erl.orig	2012-04-26 17:04:16.000000000 +0300
+++ ejabberd-2.1.11/src/web/ejabberd_http.erl	2012-07-17 12:55:07.000000000 +0300
@@ -802,7 +802,13 @@ parse_auth(_Orig = "Basic " ++ Auth64) -
 		    undefined;
 		SplitIndex ->
 		    {User, [$: | Pass]} = lists:split(SplitIndex-1, Auth),
-		    {User, Pass}
+            case string:tokens(User, "@") of
+            [Nick, Domain] -> 
+                NewUser = Nick ++ "%" ++ Domain;
+            _ ->
+                NewUser = User
+            end,
+            {NewUser ++ "@" ++ ?MYNAME, Pass}
 	    end
     end;
 parse_auth(_) ->
