--- MoinMoin/support/werkzeug/http.py.orig	2010-06-27 00:46:44.000000000 +0300
+++ MoinMoin/support/werkzeug/http.py	2011-11-30 13:58:42.000000000 +0200
@@ -441,12 +441,51 @@
                 return None  # just tell we can't parse this
 
 
+class SmartStream(object):
+    """A file-like stream that dynamically switches from memory-based
+    to file-based storage when the total amount of data is larger
+    than 500 kilobytes."""
+
+    def __init__(self, threshold=1024*500):
+        self._is_file = False
+        self._threshold = threshold
+        self._stream = StringIO()
+
+    def __getattr__(self, key):
+        return getattr(self._stream, key)
+
+    def _check(self):
+        if self._is_file:
+            return
+
+        pos = self._stream.tell()
+        if pos <= self._threshold:
+            return
+
+        stream = TemporaryFile('wb+')
+        stream.write(self._stream.getvalue())
+        stream.flush()
+        stream.seek(pos)
+
+        self._stream.close()
+        self._stream = stream
+        self._is_file = True
+
+    def write(self, *args, **kw):
+        result = self._stream.write(*args, **kw)
+        self._check()
+        return result
+
+    def writelines(self, *args, **kw):
+        result = self._stream.writelines(*args, **kw)
+        self._check()
+        return result
+
+
 def default_stream_factory(total_content_length, filename, content_type,
                            content_length=None):
     """The stream factory that is used per default."""
-    if total_content_length > 1024 * 500:
-        return TemporaryFile('wb+')
-    return StringIO()
+    return SmartStream()
 
 
 def _make_stream_factory(factory):
