*** /usr/lib/python2.4/site-packages/MoinMoin/PageEditor.py     2006-01-13 08:16:47.000000000 +0200
--- MoinMoin/PageEditor.py      2006-03-27 21:40:05.000000000 +0300
***************
*** 801,806 ****
--- 801,814 ----
                  f = open(clfn, 'w')
                  f.write(revstr+'\n')
                  f.close()
+
+                 # save to graph file, if plugin available
+                 graphsaver = wikiutil.importPlugin(self.request.cfg,
+                                                        'action',
+                                                        'savegraphdata')
+                 if graphsaver:
+                     graphsaver(self.page_name, self.request,
+                                text, pagedir, self)

                  # save to page file
                  pagefile = os.path.join(revdir, revstr)
