--- ./MoinMoin/action/newaccount.py.orig	2014-10-17 22:45:32.000000000 +0300
+++ ./MoinMoin/action/newaccount.py	2014-10-20 12:06:36.348542933 +0300
@@ -31,7 +31,8 @@
 
     # Require non-empty name
     try:
-        theuser.name = form['name']
+        name = wikiutil.clean_input(form.get('email', ['']))
+        theuser.name = name.strip()
     except KeyError:
         return _("Empty user name. Please enter a user name.")
 
@@ -104,12 +105,9 @@
 
     row = html.TR()
     tbl.append(row)
-    row.append(html.TD().append(html.STRONG().append(
-                                  html.Text(_("Name")))))
-    cell = html.TD()
-    row.append(cell)
-    cell.append(html.INPUT(type="text", size="36", name="name"))
-    cell.append(html.Text(' ' + _("(Use FirstnameLastname)")))
+    row.append(html.TD().append(html.STRONG().append(html.Text(_("Email")))))
+    row.append(html.TD().append(html.INPUT(type="text", size="36",
+                                           name="email")))
 
     row = html.TR()
     tbl.append(row)
@@ -125,12 +123,6 @@
     row.append(html.TD().append(html.INPUT(type="password", size="36",
                                            name="password2")))
 
-    row = html.TR()
-    tbl.append(row)
-    row.append(html.TD().append(html.STRONG().append(html.Text(_("Email")))))
-    row.append(html.TD().append(html.INPUT(type="text", size="36",
-                                           name="email")))
-
     textcha = TextCha(request)
     if textcha.is_enabled():
         row = html.TR()
@@ -159,7 +151,7 @@
             found = True
             break
 
-    if not found:
+    if not found and False:
         # we will not have linked, so forbid access
         request.makeForbidden(403, 'No MoinAuth in auth list')
         return
