--- ./MoinMoin/action/newaccount.py.orig	2012-02-07 23:36:47.000000000 +0200
+++ ./MoinMoin/action/newaccount.py	2012-07-29 21:58:40.249044504 +0300
@@ -31,7 +31,8 @@ def _create_user(request):
 
     # Require non-empty name
     try:
-        theuser.name = form['name']
+        name = wikiutil.clean_input(form.get('email', ['']))
+        theuser.name = name.strip()
     except KeyError:
         return _("Empty user name. Please enter a user name.")
 
@@ -103,15 +104,21 @@ def _create_form(request):
     ret.append(tbl)
     ret.append(html.Raw('</div>'))
 
+#     row = html.TR()
+#     tbl.append(row)
+#     row.append(html.TD().append(html.STRONG().append(
+#                                   html.Text(_("Name")))))
+#     cell = html.TD()
+#     row.append(cell)
+#     cell.append(html.INPUT(type="text", size="36", name="name"))
+#     cell.append(html.Text(' ' + _("(Use FirstnameLastname)")))
+
     row = html.TR()
     tbl.append(row)
-    row.append(html.TD().append(html.STRONG().append(
-                                  html.Text(_("Name")))))
-    cell = html.TD()
-    row.append(cell)
-    cell.append(html.INPUT(type="text", size="36", name="name"))
-    cell.append(html.Text(' ' + _("(Use FirstnameLastname)")))
-
+    row.append(html.TD().append(html.STRONG().append(html.Text(_("Email")))))
+    row.append(html.TD().append(html.INPUT(type="text", size="36",
+                                           name="email")))
+    
     row = html.TR()
     tbl.append(row)
     row.append(html.TD().append(html.STRONG().append(
@@ -126,12 +133,6 @@ def _create_form(request):
     row.append(html.TD().append(html.INPUT(type="password", size="36",
                                            name="password2")))
 
-    row = html.TR()
-    tbl.append(row)
-    row.append(html.TD().append(html.STRONG().append(html.Text(_("Email")))))
-    row.append(html.TD().append(html.INPUT(type="text", size="36",
-                                           name="email")))
-
     textcha = TextCha(request)
     if textcha.is_enabled():
         row = html.TR()
@@ -160,7 +161,7 @@ def execute(pagename, request):
             found = True
             break
 
-    if not found:
+    if not found and False:
         # we will not have linked, so forbid access
         request.makeForbidden(403, 'No MoinAuth in auth list')
         return
