--- ./MoinMoin/config/multiconfig.py.orig	2012-12-29 21:06:54.000000000 +0200
+++ ./MoinMoin/config/multiconfig.py	2014-11-28 10:52:58.515259865 +0200
@@ -12,6 +12,7 @@ import re
 import os
 import sys
 import time
+import imp
 
 from MoinMoin import log
 logging = log.getLogger(__name__)
@@ -45,7 +46,16 @@ def _importConfigModule(name):
     @return: module, modification time
     """
     try:
-        module = __import__(name, globals(), {})
+        configdir = None
+        for entry in sys.path:
+            if not entry:
+                continue
+            if entry.endswith('MoinMoin/support'):
+                continue
+            configdir = [entry]
+            break
+        _file, pathname, description = imp.find_module(name, configdir)
+        module = imp.load_module(name, _file, pathname, description)
         mtime = os.path.getmtime(module.__file__)
     except ImportError:
         raise
