--- ./MoinMoin/wikiutil.py.orig	2012-07-29 14:35:35.214466073 +0300
+++ ./MoinMoin/wikiutil.py	2012-07-29 14:36:15.954523034 +0300
@@ -582,6 +582,18 @@ def load_wikimap(request):
         if request.cfg.interwikiname:
             _interwiki_list[request.cfg.interwikiname] = request.getScriptname() + '/'
 
+        # collab list
+        if hasattr(request.cfg, 'collab_basedir'):
+            from collabbackend.plugin.xmlrpc.collabList import listCollabs
+            user = request.user.name
+            active = request.cfg.interwikiname
+            path = request.cfg.collab_basedir
+            baseurl = request.cfg.collab_baseurl
+            collablist = listCollabs(baseurl, user, path, active)
+            
+            for collab in collablist:
+                _interwiki_list[collab[0]] = collab[3]
+
         # save for later
         if not getattr(request.cfg.cache, 'interwiki_list', None):
             request.cfg.cache.interwiki_list = dict()
